bindings/pyroot/pythonizations/test/root_module.py

```diff
@@ -11,48 +38,71 @@ def test_import(self):
         Test import
         """
         import ROOT
-        self.assertEqual(ROOT.__name__, "ROOT")
 
+        self.assertEqual(ROOT.__name__, "ROOT")
 
     def test_relative_import(self):
         """
         Test relative import
         """
         from ROOT import TH1F
+
         self.assertEqual(TH1F.__name__, "TH1F")
         from ROOT import TH1F as Foo
+
         self.assertEqual(Foo.__name__, "TH1F")
         self.assertEqual(TH1F, Foo)
 
-
     def test_version(self):
         """
         Test __version__ property
         """
         import ROOT
+
         v = ROOT.__version__
         self.assertTrue(type(v) == str)
         self.assertIn("/", v)
         self.assertIn(".", v)
         self.assertEqual(v, ROOT.gROOT.GetVersion())
 
-
     def test_ignore_cmdline_options(self):
         """
         Test module flag to ignore command line options
         """
         import ROOT
+
         self.assertEqual(ROOT.PyConfig.IgnoreCommandLineOptions, True)
         ROOT.PyConfig.IgnoreCommandLineOptions = False
 
     def test_implicit_root_namespace(self):
         """
-        Test importing implicitely from the ROOT namespace
+        Test importing implicitly from the ROOT namespace
         """
         import ROOT
+
         ROOT.RVec
         ROOT.ROOT.RVec
         ROOT.VecOps.RVec
         ROOT.ROOT.VecOps.RVec
         ROOT.EnableImplicitMT()
         ROOT.ROOT.EnableImplicitMT()
+
+    def test_import_nested_submodules(self):
+        """
+        Test that we can correctly import C++ namespaces and other things that
+        should behave like Python modules, including nested cases.
+        """
+        #
+
+        if root_module_has("RDF.Experimental.Distributed"):
+            import ROOT.RDF.Experimental.Distributed
+
+        if root_module_has("Experimental.RNTuple"):
+            from ROOT.Experimental import RNTuple
+
+        if root_module_has("RooFit.Evaluator"):
+            from ROOT.RooFit import Evaluator
+
+
```

Comment:
Another interesting test could be import of classes defined inside of ROOT classes, e.g.

```python
from ROOT.TDirectory import TContext
```

Does `TDirectory` qualify as a Python module in this case? As in, do we want it to qualify as one or not?

Reply:
Excellent question! With my recent update, this would not work anymore because only namespaces are considered as candidates for modules, while the previous solution that replaced `__import__` didn't care.

I think we want to make PyROOT as Pythonic as possible, and in Python, one can also not use classes as modules. For example, if you have this module:
```python
# file mymodule.py

class Foo(object):

    class Bar(object):

        pass
```
You can't do `from mymodule.Foo import Bar`:
```python
import mymodule

from mymodule import Foo
from mymodule.Foo import FooBar # This doesn't work

mymodule.Foo.Bar
```
So I'd say that no Python user would expect things like that to work to begin with.

With C++ namespaces that's a different story: there is no direct equivalent in Python, and the policy that namespaces can be used as modules is intuitive to the user.

Reply:
What is your opinion on this?

Reply:
Perfect, I agree 100% with your reply!